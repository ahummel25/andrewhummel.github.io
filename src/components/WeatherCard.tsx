import React, { FC } from 'react';
import styled from '@emotion/styled';
import { format } from 'date-fns';

import { useGetFixedFahrenheit, useGetWeatherByCoords } from '../hooks';
import { breakpoints as bp } from '../styles/variables';

const Card = styled.div`
  align-self: center;
  bottom: 10px;
  padding: 8px 0 0 8px;
  position: absolute;
  border: 1px solid rgba(0, 0, 0, 0.3);
  border-radius: 12px;
  background-color: rgb(245, 245, 245);
  height: 150px;
  width: 90%;
`;

const Header = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: 10px;
`;

const WeatherImage = styled.img`
  @media (max-width: ${bp.sm}px) {
    margin-right: 8px;
  }
  width: 50px;
  height: 50px;
  margin-right: 15px;
  border-radius: 50%;
  background-color: rgb(211, 211, 211);
`;

const Title = styled.div`
  font-size: 16px;
  font-weight: bold;

  @media (min-width: ${bp.xxl}px) {
    padding-right: 3px;
  }
`;

const Temperature = styled.div`
  font-size: 16px;
  padding-bottom: 5px;
`;

const WeatherCard: FC<Record<string, unknown>> = (): JSX.Element => {
  const response = useGetWeatherByCoords();
  const icon = response?.weather[0].icon;
  const formattedDate = format(new Date(), 'EEE, \n LLL \n\n\ndo');

  return (
    <Card>
      <Header>
        <WeatherImage
          src={icon ? `https://openweathermap.org/img/w/${icon}.png` : ''}
          alt={icon}
        />
        <Title>{formattedDate}</Title>
      </Header>

      <Temperature>
        <strong>
          {useGetFixedFahrenheit(response?.main.temp_max)} &#8457;
        </strong>
        | {useGetFixedFahrenheit(response?.main.temp_min)} &#8457;
      </Temperature>

      <div>humidity: {response?.main.humidity} %</div>
      <div>speed: {response?.wind.speed} mps</div>
    </Card>
  );
};

export default WeatherCard;
