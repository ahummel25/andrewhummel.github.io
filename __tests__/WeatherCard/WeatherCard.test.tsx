import React from 'react';
import { create } from 'react-test-renderer';

import WeatherCard from '../../src/components/WeatherCard';

describe('WeatherCard', () => {
  it('renders correctly', () => {
    const tree = create(<WeatherCard />);

    expect(tree).toBeDefined();
  });
});
