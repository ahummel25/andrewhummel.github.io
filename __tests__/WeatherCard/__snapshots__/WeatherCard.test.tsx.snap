// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard renders correctly 1`] = `
.emotion-4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  bottom: 10px;
  padding: 8px 0 0 8px;
  position: absolute;
  border: 1px solid rgba(0,0,0,0.3);
  border-radius: 12px;
  background-color: rgb(245,245,245);
  height: 150px;
  width: 90%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
}

.emotion-0 {
  width: 50px;
  height: 50px;
  margin-right: 15px;
  border-radius: 50%;
  background-color: rgb(211,211,211);
}

@media (max-width:576px) {
  .emotion-0 {
    margin-right: 8px;
  }
}

.emotion-1 {
  font-size: 16px;
  font-weight: bold;
}

.emotion-3 {
  font-size: 16px;
  padding-bottom: 5px;
}

.emotion-4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  bottom: 10px;
  padding: 8px 0 0 8px;
  position: absolute;
  border: 1px solid rgba(0,0,0,0.3);
  border-radius: 12px;
  background-color: rgb(245,245,245);
  height: 150px;
  width: 90%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
}

.emotion-0 {
  width: 50px;
  height: 50px;
  margin-right: 15px;
  border-radius: 50%;
  background-color: rgb(211,211,211);
}

@media (max-width:576px) {
  .emotion-0 {
    margin-right: 8px;
  }
}

.emotion-1 {
  font-size: 16px;
  font-weight: bold;
}

.emotion-3 {
  font-size: 16px;
  padding-bottom: 5px;
}

<div
  className="emotion-4"
>
  <div
    className="emotion-2"
  >
    <img
      alt="04n"
      className="emotion-0"
      src="https://openweathermap.org/img/w/04n.png"
    />
    <div
      className="emotion-1"
    >
      Wed, Feb 24th
    </div>
  </div>
  <div
    className="emotion-3"
  >
    <strong>
      56
       ℉
    </strong>
    | 
    49
     ℉
  </div>
  <div>
    humidity: 
    82
     %
  </div>
  <div>
    speed: 
    5.82
     mps
  </div>
</div>
`;
